PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java

```diff
@@ -128,78 +140,71 @@ public SparkSubmitModel(@NotNull Project project, @NotNull SparkSubmissionParame
             }
         }
 
-        int index = submissionParameter != null ? artifactComboBoxModel.getIndexOf(submissionParameter.getArtifactName()) : -1;
+        int index = artifactComboBoxModel.getIndexOf(submissionParameter.getArtifactName());
         if (index != -1) {
             artifactComboBoxModel.setSelectedItem(submissionParameter.getArtifactName());
         }
 
         initializeTableModel(tableModel);
     }
 
+    @NotNull
     public SparkSubmissionParameter getSubmissionParameter() {
         return submissionParameter;
     }
 
-    public void setSubmissionParameters(SparkSubmissionParameter submissionParameters){
+    public void setSubmissionParameters(@NotNull SparkSubmissionParameter submissionParameters){
         this.submissionParameter = submissionParameters;
         submissionParameterMap.put(project, submissionParameter);
     }
 
+    @NotNull
     public SparkSubmitAdvancedConfigModel getAdvancedConfigModel() { return advancedConfigModel; }
 
-    public void setAdvancedConfigModel(SparkSubmitAdvancedConfigModel advancedConfigModel) {
+    public void setAdvancedConfigModel(@NotNull SparkSubmitAdvancedConfigModel advancedConfigModel) {
         this.advancedConfigModel = advancedConfigModel;
     }
 
+    public Optional<IClusterDetail> getSelectedClusterDetail() {
+        return Optional.ofNullable(mapClusterNameToClusterDetail.get(clusterComboBoxModel.getSelectedItem()));
+    }
+
+    public void setCachedClusterDetailsWithTitleMapping(List<IClusterDetail> cachedClusterDetails) {
+        this.cachedClusterDetails = cachedClusterDetails;
+
+        mapClusterNameToClusterDetail.clear();
+        cachedClusterDetails.forEach(clusterDetail -> mapClusterNameToClusterDetail.put(clusterDetail.getTitle(), clusterDetail));
+    }
+
     @NotNull
-    public IClusterDetail getSelectedClusterDetail() {
-        return mapClusterNameToClusterDetail.get((String) clusterComboBoxModel.getSelectedItem());
+    public List<IClusterDetail> getCachedClusterDetails() {
+        return cachedClusterDetails;
     }
 
-    public DefaultComboBoxModel getClusterComboBoxModel() {
+    @NotNull
+    public DefaultComboBoxModel<String> getClusterComboBoxModel() {
         return clusterComboBoxModel;
     }
 
-    public DefaultComboBoxModel getArtifactComboBoxModel() {
+    public DefaultComboBoxModel<String> getArtifactComboBoxModel() {
         return artifactComboBoxModel;
     }
 
     public boolean isLocalArtifact() {
         return submissionParameter.isLocalArtifact();
     }
 
+    @NotNull
     public Project getProject() {
         return project;
     }
 
-    public InteractiveTableModel getTableModel() {
+    public SubmissionTableModel getTableModel() {
         return tableModel;
     }
 
-    public void setClusterComboBoxModel(List<IClusterDetail> cachedClusterDetails) {
```

Comment:
>setClusterComboBoxModel [](start = 16, length = 23)

This function is not used in com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionContentPanel?

Reply:
Found the refactoring.

---
In reply to: [147347756](https://github.com/Microsoft/azure-tools-for-java/pull/1123#discussion_r147347756) [](ancestors = 147347756)